### ISSUE TITLE
Digits activity still active at the end of auth process

### ISSUE THREAD

**Issue by kaushikb1996** [crashlytics/digits-android#63]
# I KNOW THIS ISSUE HAS BEEN FIXED IN 1.9.2 release BUT THERE IS A SMALL CHANGE IN MY CASE

I am trying to build an opensource library which also needs digits for the library to work so what i did is 
i am creating the method where I am doing all the digits task for them here is the code:

```
    public void mobileLoginTest(final Intent outputActivity, final Context currentContext) throws IOException {
        String TWITTER_KEY = "MY_KEY";
        String TWITTER_SECRET = "MY_KEY";
        AuthCallback authCallback;
        final Activity activity = (Activity) currentContext;
        final TwitterAuthConfig authConfig = new TwitterAuthConfig(TWITTER_KEY, TWITTER_SECRET);
        Fabric.with(mContext, new TwitterCore(authConfig), new Digits());
        try {
            Digits.getSessionManager().clearActiveSession();
        } catch (Exception e) {
            //Do noting
        }


        authCallback = new AuthCallback() {
            @Override
            public void success(DigitsSession session, String phoneNumber) {
                if (!phoneNumber.equals(null)) {
                    mContext.startActivity(outputActivity);
                }
            }

            @Override
            public void failure(DigitsException exception) {
                try {
                    Digits.getSessionManager().clearActiveSession();
                } catch (Exception e) {
                    //Do noting
                }
            }

        };


        try {
            if(android.os.Build.VERSION.SDK_INT >= 21)
            {
                activity.finishAndRemoveTask();
            }
            else
            {
                activity.finish();
            }
        }catch (Exception e)
        {
            //Do noting
        }
        String phone = "+91";
        DigitsAuthConfig.Builder digitsAuthConfigBuilder = new DigitsAuthConfig.Builder()
                .withAuthCallBack(authCallback)
                .withPhoneNumber(phone);
        Digits.authenticate(digitsAuthConfigBuilder.build());
    }

```

But what happens here is when the auth process is completed although the digits activity is pushed to backstack and app activity is launched and also the digits activity still appears in multitasking panel and activity still exists in background and is not finished 

**And i have debugged the process and found out that auth call success method is getting executed but its not killing the Digits Activity**

and Here is the image : [http://i.pi.gy/oEWw.png](url)

**PS : I AM USING THE LATEST LIBRARY AND ALSO API LEVEL14+**

**Comment by ashwinraghav** [comment]
Hi Kaushik,
Thanks for reporting this. It is unclear to me why you would see a Digits Activity lingering. 
The activity end affinity on login success. See:
https://github.com/twitter/digits-android/blob/master/digits/src/main/java/com/digits/sdk/android/DigitsControllerImpl.java#L165

If you  can give us the android version you are testing, that might be helpful.  Also, the link of the image above seems to break. 

On an unrelated note, in the code above, you do not seem to hold a reference to the callback which might be a problem since we only hold weak references to the authCallback. See the section on "Configure the button in an activity" on https://docs.fabric.io/android/digits/digits.html

**Comment by kaushikb1996** [comment]
1) My android version on which i am testing is is  **6.0.1** on Moto X Play

2) Here is new image - `https://drive.google.com/file/d/0BzOcBarN9QzkZnNXTU9DZGlRajg/view?usp=sharing` ( URL :https://drive.google.com/file/d/0BzOcBarN9QzkZnNXTU9DZGlRajg/view?usp=sharing)

3) **I can't afford to use the digits button** so I modified the code in this manner

4) Can you please **elaborate on holding reference to callback**

**Comment by kaushikb1996** [comment]
@ashwinraghav Guys still waiting for the answer

**Comment by srinivasanj** [comment]
I have the exact same issue...

Android 6.0 with...

> compile('com.digits.sdk.android:digits:1.10.3@aar') {
>         transitive = true;
>     }

**Comment by ashwinraghav** [comment]
We have been unable to repro this issue at our end. Do you think you can upgrade Digits versions after 1.11.2 and let us know whether you are seeing the issue? It might be worth investigating since there seems to be a few people seeing this. Thanks for the help.

**Comment by lookitsatravis** [comment]
Howdy, all. We are using v2.0.0 and running into this issue as well. The primary device I've been testing with is running 5.1.1 (Samsung Galaxy J1). In my case, the digits activity is still running when:

* The back button is clicked to hide the keyboard and then return to the main activity
* The auth workflow is complete and focus is returned to the main activity

In either case, the Digits activity remains active in a separate window

**Comment by RMMHK** [comment]
Hello all respective members, i'm using digits 2.0.0 version and from last 2 weeks i'm trying to find why digits 2.0.0 isn't allowing me to start new activity from onSuccess method.. Please help me , i'm very stressed. :( :(

**Comment by srMarlins** [comment]
I'm seeing the same issue as @lookitsatravis except I'm using v2.0.1 and my main testing device is 6.0.1 (Nexus 5)

**Comment by mouli1382** [comment]
I am seeing the same issue on my Nexus 5 running 6.0.1 with digits version 2.0.2

**Comment by mehernosh** [comment]
My two cents:
I found a similar problem while using Digits in cordova.

The root cause was [around here](https://github.com/twitter/digits-android/blob/master/digits/src/main/java/com/digits/sdk/android/DigitsClient.java#L156).
I discovered that Fabric had not been initialized in time and the statement `digits.getFabric().getCurrentActivity() ` was returning a `null`.

Thus, the Digits activity is being started with the intent flags : `Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP` . 
I solved this in the context of my project by making sure `Fabric.with()` is called much before any calls to Digits.authenticate.
